src/Chapter1.hs

```diff
@@ -554,7 +546,15 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x <= z && x >= y = x
+    | x <= y && x >= z = x
+    | y <= z && y >= x = y
+    | y <= x && y >= z = y
+    | z <= x && z >= y = z
+    | z <= y && z >= x = z
+    | otherwise = z
```

Comment:
It looks like you can move all these 3 cases under a single otherwise and return z in here.
The implementation will be a bit shorter 🙂

As you noticed, you have covered all possible cases, so if it doesn't fit in the first 4 cases, then it will definitely satisfy one of these, but we don't care which one, we don't even need to check, as we know that the answer is going to be `z`! ⚡